### PR TITLE
set imcalc default option data type

### PR DESCRIPTION
**Pull request by CerenB**
Related to the issue #365

### CODE REVIEW

**Comment by CerenB** [comment]
I will re-open the issue when the skullstripping-demo PR #353 is accepted. I built this PR on another PR...

**Comment by Remi-Gau** [comment]
I updated this PR.

Just made sure that we can more easily decided what data type we want to use.

Also made some changes to start using the input parser.

**Comment by Remi-Gau** [comment]
Not sure why the tests are not running with github action but they pass locally.

File: src/batches/setBatchImageCalculation.m [hunk 1]
```diff
@@ -16,22 +16,70 @@
...
+  allowedDataType = {'uint8', ...
+                     'int16', ...
+                     'int32', ...
+                     'float32', ...
+                     'float64', ...
+                     'int8', ...
+                     'uint16', ...
+                     'uint32'};
+
+  defaultDataType = 'int16';
```
Comment (Remi-Gau) [inline src/batches/setBatchImageCalculation.m:51]: @CerenB you had set the value to `16` which would correspond to `'float32'`.

Should I change the default here?

Comment (CerenB) [reply src/batches/setBatchImageCalculation.m:51]: ah I see now! I do not know how/where would you like to call `defaultDataType` ?  Would we call it somewhere that value like `checkOptions.m` ?

For me, I need to set it to float32. Not sure for other users how it should be set.

Comment (Remi-Gau) [reply src/batches/setBatchImageCalculation.m]: that's fine to just have the default in that file the way you changed it

if we run into trouble with that later, we have a way to access it via the input argument so we should be fine.

**Review by Remi-Gau — APPROVED** [review]

**Pull request merged**